_format/1.1/index.md

```diff
@@ -62,8 +59,22 @@ a request specifies the header `Content-Type: application/vnd.api+json`
 with any media type parameters other than `profile`.
 
 Servers **MUST** respond with a `406 Not Acceptable` status code if a
-request's `Accept` header contains the JSON:API media type and all instances
-of that media type are modified with media type parameters.
+request's `Accept` header contains the JSON:API media type and the server is
+unable to respond with an acceptable representation. The response **MUST** contain
+an [error object] which lists the unsupported profile or profiles as
+its `source` and which has the following URI as (one of) its `type`s:
+
+```
+https://jsonapi.org/errors/profile-not-supported
+```
+
+It is **RECOMMENDED** that servers specify the `Vary` header with `Accept`
+as one of its header names. This will ensure that the server's responses are
+appropriately cached and future client/server negotiations will work as
+expected.
```

Comment:
Perhaps this recommendation ought to be elevated to a **SHOULD**?

Servers that do not support any profiles or only a specific set of profiles probably should not set the `Vary` header because it opens them up to the possibility of a malicious actor sending many variations of the `Accept` header to fill the server's cache.

OTOH, if negotiation is supported, not setting the `Vary` header will break profile negotiation whenever a cache is used.

Reply:
@wimleers, I think you _were_ right about *MUST*. However, since you made this comment, the direction of profiles has been changed such that a MUST would probably be too heavy handed. IOW, it's valid for a client to ask for profile A, but get a cached response with profile B. I'm gonna change this to a *SHOULD*.

A subsequent PR will add extension negotiation, which is not optional, where a *MUST* will be appropriate.